geomesa-hbase/geomesa-hbase-datastore/src/main/scala/org/locationtech/geomesa/hbase/data/HBaseIndexAdapter.scala

```diff
@@ -130,15 +131,23 @@ class HBaseIndexAdapter(ds: HBaseDataStore) extends IndexAdapter[HBaseDataStore]
         }
       }
 
-      // wait for the table to come online
-      if (!admin.isTableAvailable(name)) {
-        val timeout = TableAvailabilityTimeout.toDuration.filter(_.isFinite())
-        logger.debug(s"Waiting for table '$table' to become available with " +
-            s"${timeout.map(t => s"a timeout of $t").getOrElse("no timeout")}")
-        val stop = timeout.map(t => System.currentTimeMillis() + t.toMillis)
-        while (!admin.isTableAvailable(name) && stop.forall(_ > System.currentTimeMillis())) {
-          Thread.sleep(1000)
-        }
+      waitForTable(admin, name)
+    }
+  }
+
+  override def renameTable(from: String, to: String): Unit = {
+    WithClose(ds.connection.getAdmin) { admin =>
+      val existing = TableName.valueOf(from)
+      val renamed = TableName.valueOf(to)
+      if (admin.tableExists(existing)) {
+        // renaming in hbase requires creating a snapshot and using that to create the new table
+        val snapshot = StringSerialization.alphaNumericSafeString(UUID.randomUUID().toString)
+        admin.disableTable(existing)
+        admin.snapshot(snapshot, existing)
+        admin.cloneSnapshot(snapshot, renamed)
```

Comment:
Do we want any error handling around this? TableExistsException or RestoreSnapshotException

Reply:
I think generally we just let the exceptions propagate so that the user can handle them. Is there a good way to handle errors and recover here? `TableExistsException` should be unlikely as it uses a UUID (assuming the exception would come from `snapshot` already existing?)

Reply:
I was thinking table exists would come from the renamed value being bad or existing == renamed.

Reply:
oh, yeah that makes sense. do you have any suggestions for handling errors, or think it's ok to propagate them?